core/src/main/scala/org/apache/spark/sql/execution/datasources/StoreDataSourceStrategy.scala

```diff
@@ -202,4 +205,137 @@ private[sql] object StoreDataSourceStrategy extends Strategy {
         filterCondition.map(execution.FilterExec(_, scan)).getOrElse(scan))
     }
   }
+
+  /**
+   * Tries to translate a Catalyst [[Expression]] into data source [[Filter]].
+   *
+   * @return a `Some[Filter]` if the input [[Expression]] is convertible, otherwise a `None`.
+   */
+  protected[sql] def translateFilter(predicate: Expression): Option[Filter] = {
```

Comment:
I suppose this is a copy of some Spark code? Mention the same in the comment of this method "adapted from Spark's ..." and also add the Spark copyright header to this file.